### PR TITLE
(feat-rocksdb) Remove Transaction and TransactionSource

### PR DESCRIPTION
**Pull request by dwerner**
This PR addresses tech debt remaining in feat-rocksdb by removing the Transaction and TransactionSource traits, refactoring affected test code.

### CODE REVIEW

File: execution_engine/src/storage/trie_store/db.rs [hunk 1]
```diff
@@ -200,16 +145,215 @@ impl Store<Digest, Trie<Key, StoredValue>> for ScratchTrieStore {
     }
 }
 
-impl TrieStore<Key, StoredValue> for ScratchTrieStore {}
+/// Represents the state of a migration of a state root from lmdb to rocksdb.
+pub enum RootMigration {
+    /// Has the migration been left incomplete
+    NotMigrated,
+    /// Has the migration been not yet been started or completed
+    Incomplete,
```
Comment (Fraser999) [inline execution_engine/src/storage/trie_store/db.rs]: Maybe [`Ongoing` or `Partial`] and `NotStarted` would be clearer than `NotMigrated` and `Incomplete` respectively?

**Review by Fraser999 — APPROVED** [review]
One non-blocking suggestion, otherwise LGTM.

File: execution_engine/src/storage/trie_store/db.rs (hunk 1 above)
Comment (dwerner) [reply execution_engine/src/storage/trie_store/db.rs]: Applied in https://github.com/casper-network/casper-node/pull/2984/commits/59622e0dc21b45405b6acca64a39900ad6fef28e

File: execution_engine/src/storage/store/mod.rs [hunk 2]
```diff
@@ -5,30 +5,36 @@ pub(crate) mod tests;
...
+/// Base trait for db operations that can raise an error.
+pub trait ErrorSource: Sized {
+    /// Type of error held by the trait.
     type Error: From<bytesrepr::Error>;
+}
 
-    /// Underlying store type.
-    type Handle;
+/// Represents a store that can be read from.
+pub trait Readable: ErrorSource {
```
Comment (goral09) [inline execution_engine/src/storage/store/mod.rs:16]: Why did you change from type member to inheritence?

File: execution_engine/src/storage/trie_store/db.rs [hunk 3]
```diff
@@ -2,76 +2,48 @@
...
-    fn handle(&self) -> Self::Handle {
-        self.db
-    }
-}
+/// Column family name for tracking progress of data migration.
+const LMDB_MIGRATED_STATE_ROOTS_COLUMN_FAMILY: &str = "lmdb_migrated_state_roots_column";
 
-impl<K, V> TrieStore<K, V> for LmdbTrieStore {}
+/// Working set column famliy.
+const WORKING_SET_COLUMN_FAMILY: &str = "working_set";
```
Comment (goral09) [inline execution_engine/src/storage/trie_store/db.rs]: This is used only in a method that isn't used itself. Also, this change isn't required for _removal of `Trasnaction` and `TransactionSource`_

File: execution_engine/src/storage/trie_store/db.rs [hunk 4]
```diff
@@ -2,76 +2,48 @@
 
 use std::{
     collections::HashMap,
+    path::{Path, PathBuf},
     sync::{Arc, Mutex},
 };
 
-use casper_types::{bytesrepr, Key, StoredValue};
-use lmdb::{Database, DatabaseFlags};
+use rocksdb::{
```
Comment (goral09) [inline execution_engine/src/storage/trie_store/db.rs:9]: There's a comment in line 1 that mentions LMDB.

File: node/src/components/contract_runtime/config.rs [hunk 5]
```diff
@@ -1,68 +1,29 @@
...
     /// Defaults to 5.
     max_query_depth: Option<u64>,
-    /// Enable synchronizing to disk only after each block is written.
-    ///
-    /// Defaults to `false`.
-    enable_manual_sync: Option<bool>,
 }
 
 impl Config {
-    pub(crate) fn max_global_state_size(&self) -> usize {
```
Comment (goral09) [inline node/src/components/contract_runtime/config.rs:37]: RocksDB doesn't need this config?

File: node/src/reactor/participating.rs [hunk 6]
```diff
@@ -733,20 +733,24 @@ impl reactor::Reactor for Reactor {
...
                             "unable to retrieve parent block at height {} for migration to rocksdb {:?}",
                             height, err,
                         );
                         continue;
                     }
                 };
+                info!(
+                    "Will migrate state root {:#?} at height {}",
+                    state_roots, height
+                );
```
Comment (goral09) [inline node/src/reactor/participating.rs]: if this is using the tracing crate then it should be:
```rust
info!(?state_roots, ?height, "will migrate state root at height")
```

File: execution_engine/src/storage/store/mod.rs (hunk 2 above)
Comment (dwerner) [reply execution_engine/src/storage/store/mod.rs:16]: This was needed to handle everywhere that Transaction's Self::Error type was referenced. I tried adding Error as an associated type but it wasn't as compatible with our existing top level functions (`write`, `put_stored_values` etc) trait bounds, so I settled on this approach.

File: node/src/components/contract_runtime/config.rs (hunk 5 above)
Comment (dwerner) [reply node/src/components/contract_runtime/config.rs:37]: Correct.

File: node/src/reactor/participating.rs (hunk 6 above)
Comment (dwerner) [reply node/src/reactor/participating.rs]: Thanks. And now that I review this as well, this is actually logging the entire debug representation of the state_roots vec, so I have changed this to only log the exact state root.

Comment (dwerner) [reply node/src/reactor/participating.rs]: Fixed in 213bf27a50f00424def44d1557567f3f3d184e21

File: execution_engine/src/storage/trie_store/db.rs (hunk 3 above)
Comment (dwerner) [reply execution_engine/src/storage/trie_store/db.rs]: Good catch. Fixed in 213bf27a50f00424def44d1557567f3f3d184e21

File: execution_engine/src/storage/trie_store/db.rs (hunk 4 above)
Comment (dwerner) [reply execution_engine/src/storage/trie_store/db.rs:9]: Fixed in 213bf27a50f00424def44d1557567f3f3d184e21

**Review by goral09 — APPROVED** [review]
